File: typescript/src/models/checkoutSessionCreateRequest.ts
// this file is @generated
import { type Address, AddressSerializer } from "./address";
import { type Buyer, BuyerSerializer } from "./buyer";
import { type Item, ItemSerializer } from "./item";

export interface CheckoutSessionCreateRequest {
  buyer?: Buyer | null;
  items: Item[];
  fulfillmentAddress?: Address | null;
}

export const CheckoutSessionCreateRequestSerializer = {
  _fromJsonObject(object: any): CheckoutSessionCreateRequest {
    return {
      buyer: object["buyer"]
        ? BuyerSerializer._fromJsonObject(object["buyer"])
        : undefined,
      items: object["items"].map((item: Item) => ItemSerializer._fromJsonObject(item)),
      fulfillmentAddress: object["fulfillment_address"]
        ? AddressSerializer._fromJsonObject(object["fulfillment_address"])
        : undefined,
    };
  },

  _toJsonObject(self: CheckoutSessionCreateRequest): any {
    return {
      buyer: self.buyer ? BuyerSerializer._toJsonObject(self.buyer) : undefined,
      items: self.items.map((item) => ItemSerializer._toJsonObject(item)),
      fulfillment_address: self.fulfillmentAddress
        ? AddressSerializer._toJsonObject(self.fulfillmentAddress)
        : undefined,
    };
  },
};

File: rust/src/models/checkout_session_create_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{address::Address, buyer::Buyer, item::Item};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct CheckoutSessionCreateRequest {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub buyer: Option<Buyer>,
    pub items: Vec<Item>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub fulfillment_address: Option<Address>,
}
